R/defineModel.R

```diff
@@ -1,18 +1,126 @@
+                        compute.fit = TRUE, pvMethod = c("regular", "bayesian"),
+                        fitTamMmlForBayesian = TRUE, n.plausible=5, seed = NULL,
+                        conquest.folder=NULL, constraints=c("cases","none","items"),
+                        std.err=c("quick","full","none"), distribution=c("normal","discrete"),
+                        method=c("gauss", "quadrature", "montecarlo", "quasiMontecarlo"),
+                        n.iterations=2000, nodes=NULL, p.nodes=2000, f.nodes=2000,
+                        converge=0.001, deviancechange=0.0001, equivalence.table=c("wle","mle","NULL"),
+                        use.letters=FALSE, allowAllScoresEverywhere = TRUE, guessMat = NULL,
+                        est.slopegroups = NULL, fixSlopeMat = NULL, slopeMatDomainCol=NULL,
+                        slopeMatItemCol=NULL, slopeMatValueCol=NULL, progress = FALSE,
+                        Msteps = NULL, increment.factor=1 , fac.oldxsi=0,
+                        export = list(logfile = TRUE, systemfile = FALSE, history = TRUE,
+                                      covariance = TRUE, reg_coefficients = TRUE, designmatrix = FALSE) ){
+
+### checking/asserting the arguments -------------------------------------------
+  # dat
+  ismc <- attr(dat, "multicore")
+  dat  <- eatTools::makeDataFrame(dat, name = "dat")
+
+  # data frame: qMatrix
+  checkmate::assert_data_frame(qMatrix, null.ok = TRUE, col.names = "named", min.cols = 2)
+  warning(paste0("The first column of the data frame `qMatrix` should be called `item`, but is called ",
+                 colnames(qMatrix)[1], "."))
+
+  # dataframe: anchor
+  checkmate::assert_data_frame(anchor, null.ok = TRUE, col.names = "named", min.cols = 2)
+  if(checkmate::test_data_frame(anchor, null.ok = TRUE, col.names = "named", min.cols = 3)){
+    checkmate::assert_subset(colnames(anchor), choices = c(colnames(anchor)[1:2],
+                                                           "domainCol", "itemCol", "valueCol")) }
+
+  # list: splitted Models
+  checkmate::assert_list(splittedModels, null.ok = TRUE)
+  # subset: irtmodel
+  checkmate::assert_subset(irtmodel, choices = c("1PL", "2PL", "PCM", "PCM2","RSM",
+                                                 "GPCM", "2PL.groups", "GPCM.design", "3PL"))
+
+  #' assert vector:
+  #' items, id, DIF.var, HG.var, group.var, weight.var, schooltype.var
+  checkmate::assert_vector(items)
+  lapply(c(id, DIF.var), checkmate::assert_vector, len = 1)
+  lapply(c(HG.var, group.var, weight.var), checkmate::assert_vector, null.ok = TRUE)
+  checkmate::assert_vector(schooltype.var, len = 1, null.ok = TRUE)
+
+  #' assert numeric:
+  #' minNperItem, boundary, n.plausible, seed, n.iterations, nodes, converge,
+  #' deviancechange, Msteps
+  lapply(c(minNperItem, boundary, n.plausible, n.iterations, nodes, converge,
+           deviancechange, Msteps), checkmate::assert_numeric, len = 1)
```

Comment:
I don't know whether we might differentiate between only-positive integers (minNperItem, boundary, n.plausible, n.iterations, nodes, Msteps, etc.) and numerical values?